alibi_detect/saving/_pytorch/saving.py

```diff
@@ -0,0 +1,131 @@
+            # embedding
+            embed = layers[0]
+            cfg_embed = save_embedding_config(embed, base_path, local_path.joinpath('embedding'))
+            # preprocessing encoder
+            model = layers[1]
+        else:  # If UAE is simply an encoder
+            model = model.encoder
+    elif isinstance(model, TransformerEmbedding):
+        cfg_embed = save_embedding_config(model, base_path, local_path.joinpath('embedding'))
+        model = None
+    elif isinstance(model, HiddenOutput):
+        model = model.model
+    elif isinstance(model, nn.Module):  # Last as TransformerEmbedding and UAE are nn.Module's
+        model = model
+    else:
+        raise ValueError('Model not recognised, cannot save.')
+
+    if model is not None:
+        filepath = base_path.joinpath(local_path)
+        save_model(model, filepath=filepath)
+        cfg_model = {
+            'flavour': Framework.PYTORCH.value,
+            'src': local_path.joinpath('model')
+        }
+    return cfg_model, cfg_embed
+
+
+def save_model(model: nn.Module,
+               filepath: Union[str, os.PathLike],
+               save_dir: Union[str, os.PathLike] = 'model') -> None:
+    """
+    Save PyTorch model.
+
+    Parameters
+    ----------
+    model
+        The PyTorch model to save.
+    filepath
+        Save directory.
+     save_dir
+        Name of folder to save to within the filepath directory.
+    """
+    # create folder to save model in
+    model_path = Path(filepath).joinpath(save_dir)
+    if not model_path.is_dir():
+        logger.warning('Directory {} does not exist and is now created.'.format(model_path))
+        model_path.mkdir(parents=True, exist_ok=True)
+
+    # save model
+    model_path = model_path.joinpath('model.pt')
+
+    if isinstance(model, nn.Module):
+        torch.save(model, model_path, pickle_module=dill)
+    else:
+        raise ValueError('The extracted model to save is not a `nn.Module`. Cannot save.')
+
+
+def save_embedding_config(embed: TransformerEmbedding,
+                          base_path: Path,
+                          local_path: Path = Path('.')) -> dict:
+    """
+    Save embeddings for text drift models.
+
+    Parameters
+    ----------
+    embed
+        Embedding model.
+    base_path
+        Base filepath to save to (the location of the `config.toml` file).
+    local_path
+        A local (relative) filepath to append to base_path.
+    """
+    # create folder to save model in
+    filepath = base_path.joinpath(local_path)
+    if not filepath.is_dir():
+        logger.warning('Directory {} does not exist and is now created.'.format(filepath))
+        filepath.mkdir(parents=True, exist_ok=True)
+
+    # Populate config dict
+    cfg_embed = {}  # type: Dict[str, Any]
+    cfg_embed.update({'type': embed.emb_type})
+    cfg_embed.update({'layers': embed.hs_emb.keywords['layers']})
+    cfg_embed.update({'src': local_path})
+    cfg_embed.update({'flavour': Framework.PYTORCH.value})
```

Comment:
Does it make sense to just have a `get_config` method on the embedding model? like how we do with the [tensorflow kernel](https://github.com/SeldonIO/alibi-detect/blob/57488644af894a7e4e528718c7d4ef86156ee2fa/alibi_detect/utils/tensorflow/kernels.py#L91).

Reply:
hmm, although looking again at [this](https://github.com/SeldonIO/alibi-detect/blob/57488644af894a7e4e528718c7d4ef86156ee2fa/alibi_detect/utils/tensorflow/kernels.py#L91) i've realised that these things aren't exactly similar... the tf kernel `get_config` overide is just used to convert `sigma` from `torch.tensor` to a `numpy` array. I guess what I mean is would it make sense to move the config stuff to the embedding object instead of doing it here?